File: github-gitlab.py
import os
import sys
import datetime

# Obtener el mensaje del commit y la rama (si se proporcionan)
commit_msg = sys.argv[1] if len(sys.argv) > 1 else datetime.datetime.now().strftime("Commit del %Y-%m-%d %H:%M:%S")
branch = sys.argv[2] if len(sys.argv) > 2 else 'main'

# Ejecutar los comandos de git
os.system('git add .')
os.system(f'git commit -m "{commit_msg}"')
os.system(f'git push github {branch}')
os.system(f'git push gitlab {branch}')

# import argparse
# import os
#
# parser = argparse.ArgumentParser()
# parser.add_argument('message', help='Mensaje de commit')
# args = parser.parse_args()
#
# # Ejecutar los comandos de git
# os.system('git add .')
# os.system(f'git commit -m "{args.message}"')
# os.system('git push github main')
# os.system('git push gitlab main')

# import os
#
# # Ejecutar los comandos de git
# os.system('git add .')
# os.system('git commit -m "Primer commit"')
# os.system('git push github main')
# os.system('git push gitlab main')

# OTRA MANERA DE HACERLO
# import os
#
# # Ejecutar los comandos de git
# for cmd in ['git add .', 'git commit -m "Primer commit"', 'git push github main', 'git push gitlab main']:
#     output = os.popen(cmd).read()
#     print(output)

File: github-gitlab.js
const { spawnSync } = require('child_process');
const { format } = require('date-fns');

// Obtener la fecha y hora actual en formato deseado
const now = format(new Date(), "dd 'de' MMMM 'de' yyyy - HH:mm:ss", { locale: require('date-fns/locale/es') });

// Mensaje por defecto si no se proporciona uno
const defaultCommitMessage = `Commit realizado el ${now}`;

// Obtener el mensaje y la rama de los argumentos, si se proporcionan
const args = process.argv.slice(2);
const commitMessageIndex = args.indexOf('-m');
const branchIndex = args.indexOf('-b');
const commitMessage = commitMessageIndex !== -1 ? args[commitMessageIndex + 1] : defaultCommitMessage;
const branch = branchIndex !== -1 ? args[branchIndex + 1] : 'main';

// Ejecutar los comandos de git
spawnSync('git', ['add', '.']);
spawnSync('git', ['commit', '-m', commitMessage]);
spawnSync('git', ['push', 'github', branch]);
spawnSync('git', ['push', 'gitlab', branch]);
// spawnSync('git', ['push', 'heroku', branch]);
// spawnSync('heroku', ['open']);
